### ISSUE TITLE
srs edge模式下能否按需回源, edge standby

### ISSUE THREAD

**Issue by backylee** [ossrs/srs#419]
srs edge模式,能否在源站需要流的时候再push到源站,减少带宽浪费.

目前的srs：回源取流（播放时）或者将流转发 给源站（=>发布时<=）
按需回源的srs：回源取流（播放时）或者将流转发 给源站（=>源站播放时<=）

【盟主】可爱的雷万宝 2015/6/5 16:45:16
只不过 SRS 里面的 配置是 写死的 。。相当于  静态路由
16:47:54
【盟主】winlin 2015/6/5 16:47:54
加个standby功能，提个bug吧

【宗师】湖南-前台-小白
举个简单的栗子。。。某个单位有100路监控，他们有个指挥中心，只实时播放16路流

【宗师】BK 2015/6/5 16:49:20
这个栗子非常好

实际场景:
客户厂区有1000路摄像头,发布到厂区的edge服务器里,实时录像.
源服务器对外公开播放流,提供厂区参观,3D厂区展示等业务。源站不止服务一个厂区，如果源站服务10个厂区，1w路流，这时候源站不希望那1000路全回源，有play的时候再到源站来。以减少源站的压力。

**Comment by winlinvip** [comment]
Can you provide real examples? You can omit key client information.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Explain the difficulties you are currently facing, without any assumptions about why the user needs this feature. It should be something like this: describe the problem you are currently facing, what the customer needs, what you need to provide, where you are stuck, and how you have managed to work around it. It would be convenient if the SRS (Software Requirements Specification) could support certain aspects. Something along these lines.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
This can be called: edge standby mode, waiting for the origin server to request streaming before starting to stream.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
The previous mode was RTSP, with each server being a source station. Now, after switching to the SRS cluster, the stream is pushed to the edge and delivered directly to the source station; unlike the streams played at the edge, which are sourced on demand.

`TRANS_BY_GPT3`

**Comment by lanybass** [comment]
Waiting for implementation. This feature is too basic. I can't believe it hasn't been done, have they not considered the bandwidth?

The situation I encountered is similar to the example above. I have to provide access to hundreds of cameras for a certain boss to view at any time, but most of the time, he doesn't even look at them. Occasionally, he might only view one of them. Although it is said that it takes a long time to raise an army but only a moment to use it, sending several hundred video streams from his internal network to my public network server is completely unsustainable. If it is pushed on-demand, it would be more reasonable to temporarily push the stream of the camera he wants to view.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Waiting for PR.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
There is one implementation method where, during standby, after the edge connects to the origin server, it only sends ping packets to maintain the connection, or sends control message packets, or empty packets, in short, it keeps the data volume from the edge to the origin server very small, thus supporting a very large number of connections. When there is a need for source playback, the origin server sends control messages to the edge, and the edge then delivers the stream.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Looking back at this feature, it is a requirement for monitoring scenes. Generally, there will be additional signaling. After the signaling interaction for watching or adjusting the camera position is completed, the stream will be sent to the media server.

This requirement is merged into #1500 gb18181 for support.

`TRANS_BY_GPT3`
